library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java

```diff
@@ -1214,14 +1215,38 @@ private static int getThemeAccentColor(Context context) {
      * @see java.util.Calendar
      */
     public void setFirstDayOfWeek(int day) {
-        adapter.setFirstDayOfWeek(day);
+        firstDayOfWeek = day;
+        CalendarPagerAdapter<?> newAdapter;
```

Comment:
Add TODO to consider a less nuclear means of resetting the adapter when setting a new first day of week and how regular notifyDataSetChanged doesn't work (may require updating getItemPosition to flag current object and the ones to the left/right as changed)